test/node/int_32_tests.js

```diff
@@ -97,4 +98,50 @@ describe('Int32', function () {
       });
     }
   });
+
+  describe('fromString', () => {
+    const acceptedInputs = [
+      ['Int32.max', '2147483647', 2147483647],
+      ['Int32.min', '-2147483648', -2147483648],
+      ['zero', '0', 0],
+      ['non-leading zeros', '45000000', 45000000],
+      ['zero with leading zeros', '000000', 0],
+      ['positive leading zeros', '000000867', 867],
+      ['explicity positive leading zeros', '+000000867', 867],
+      ['negative leading zeros', '-00007', -7]
+    ];
+    const errorInputs = [
+      ['Int32.max + 1', '2147483648'],
+      ['Int32.min - 1', '-2147483649'],
+      ['positive integer with decimal', '2.0'],
+      ['zero with decimal', '0.0'],
+      ['negative zero', '-0'],
+      ['Infinity', 'Infinity'],
+      ['-Infinity', '-Infinity'],
+      ['NaN', 'NaN'],
+      ['fraction', '2/3'],
+      ['commas', '34,450'],
+      ['exponentiation notation', '1e1'],
+      ['octal', '0o1'],
+      ['binary', '0b1'],
+      ['hex', '0x1'],
+      ['empty string', ''],
+      ['leading and trailing whitespace', '    89   ', 89]
```

Comment:
According to the AC on the ticket, leading and trailing whitespace should be allowed

Reply:
Prohibiting whitespace is a change Neal and I discussed in office. This is because we want to stay consistent to the `decimal128.fromString` implementation. I can also ask the team for more input!

Reply:
🤷 conversely, Long.fromString does support leading and trailing whitespace:

```javascript
> bson.Long.fromString('   5')
new Long('5')
> bson.Long.fromString('       5')
new Long('5')
> bson.Long.fromString('   10   ')
new Long('10')
```

Reply:
imo, it's better to be more strict now since this is a public function, and then if we decide to allow it later, we can. 
Also, `Long.fromString` is a function we're likely going to change for v7, while `decimal128.fromString` is more stable.